### PR TITLE
feat(askui-nodejs): allow tracking time to value using install timest…

### PR DESCRIPTION
**Pull request by adi-wan-askui**
…amp (#AS-1070)

### CODE REVIEW

File: packages/askui-nodejs/src/utils/analytics/installation-timestamp.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return InstallationTimestamp.value;
+  }
+
+  private static async getFromFile(): Promise<Date> {
+    return new Promise((resolve, reject) => {
+      fs.readFile(InstallationTimestamp.fileName, 'utf-8', (err, isoDateStr) => {
+        if (err) {
+          reject(new InstallationTimestampGetError(err));
+        } else {
+          resolve(new Date(isoDateStr));
```
Comment (mlikasam-askui) [inline packages/askui-nodejs/src/utils/analytics/installation-timestamp.ts:43]: should we also check if isoDateStr is valid?

File: packages/askui-nodejs/bin/askui-postinstall [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env node
+
+const { access } = require('fs/promises');
+const { join } = require('path');
+
+async function shouldBeExecuted() {
+  try {
+    await access(join(__dirname, '../.no-askui-postinstall'));
```
Comment (mlikasam-askui) [inline packages/askui-nodejs/bin/askui-postinstall:8]: is there a specific reason why "access" is used here and not "exists"?

Comment (adi-wan-askui) [reply packages/askui-nodejs/bin/askui-postinstall:8]: `exists` is marked for deprecation and should generally not be used, see Node.js docs

File: packages/askui-nodejs/src/utils/analytics/installation-timestamp.ts (hunk 1 above)
Comment (adi-wan-askui) [reply packages/askui-nodejs/src/utils/analytics/installation-timestamp.ts:43]: It doesn't make sense from my perspective. It is unlikely that a user is going to mess with the contents of that file except people trying to misuse our api but we cannot prevent them from sending us bad data at this stage anyway. Let's say someone really changed it by accident, then, If the string was really not interpretable as a date, an error would be raised that would be caught and we would not get an installation timestamp.

**Review by mlikasam-askui — APPROVED** [review]

**Pull request merged**